### PR TITLE
JDBC driver implements getDate, getTimestamp and getTime with Calendar in ResultSet

### LINKED ISSUE #5650: [TASK][EASY] JDBC driver implements `Result#getDate` with `Calendar` parameters
### What's the level of this task?

EASY

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before creating

- [X] I have searched in the [task list](https://github.com/orgs/apache/projects/296) and found no similar tasks.

### Mentor

- [ ] I have sufficient expertise on this task, and I volunteer to be a mentor of this task to guide contributors through the task.

### Skill requirements

- JDBC concepts
- Java

### Background and Goals

https://github.com/apache/kyuubi/discussions/5649

### Implementation steps

Implement the required methods, and test with Tableau

### Additional context

Introduction of [2023 Kyuubi Code Contribution Program](https://github.com/apache/kyuubi/issues/5357)

### PR DESCRIPTION
**Pull request by zhaohehuhu**
# :mag: Description
## Issue References 🔗

This pull request fixes #5650

## Describe Your Solution 🔧

 implemented getDate, getTimestamp and getTime with Calendar in Kyuubi JDBC driver side

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [ ] Pull request title is okay.
- [ ] No license issues.
- [ ] Milestone correctly set?
- [ ] Test coverage is ok
- [ ] Assignees are selected.
- [ ] Minimum number of approvals
- [ ] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

**Comment by zhaohehuhu** [comment]
@pan3793  plz review this feature. Thanks!

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiArrowBasedResultSet.java [hunk 1]
```diff
@@ -198,6 +199,32 @@ public Date getDate(String columnName) throws SQLException {
     return getDate(findColumn(columnName));
   }
 
+  public Date getDate(int columnIndex, Calendar cal) throws SQLException {
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiArrowBasedResultSet.java:202]: please add `@Overwrite`, and can we remove it from the parent abstract class?

**Comment by cfmcgrady** [comment]
would you mind adding UTs within `org.apache.kyuubi.operation.SparkDataTypeTests`?

**Comment by zhaohehuhu** [comment]
> would you mind adding UTs within `org.apache.kyuubi.operation.SparkDataTypeTests`?
Of course not. Let me take a look. Thanks!

**Comment by zhaohehuhu** [comment]
> SparkDataTypeTests

added. plz review. Thanks.

**Comment by cfmcgrady** [comment]
cc  @cxzl25

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/SparkDataTypeTests.scala [hunk 2]
```diff
@@ -213,6 +231,26 @@ trait SparkDataTypeTests extends HiveJDBCTestHelper with SparkVersionUtil {
...
   }
 
+  test("execute statement - select timestamp - second with calendar") {
+    withJdbcStatement() { statement =>
+      val resultSet = statement.executeQuery(
+        "SELECT TIMESTAMP '2018-11-17 13:33:33' AS col")
+      assert(resultSet.next())
+      assert(resultSet.getTimestamp(
+        "col",
+        Calendar.getInstance(TimeZone.getTimeZone("UTC"))) === Timestamp.valueOf(
```
Comment (pan3793) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/operation/SparkDataTypeTests.scala:241]: what if the TimeZone is not UTC?

Comment (zhaohehuhu) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/operation/SparkDataTypeTests.scala]: That would be a problem. Refactored.

Comment (zhaohehuhu) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/operation/SparkDataTypeTests.scala]: Let me check it.

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java [hunk 3]
```diff
@@ -412,6 +440,85 @@ public Timestamp getTimestamp(String columnName) throws SQLException {
...
+    Time value = getTime(columnIndex);
+    if (value == null) {
+      return null;
+    } else {
+      try {
+        return parseTime(value, cal);
+      } catch (IllegalArgumentException e) {
+        throw new KyuubiSQLException("Cannot convert column " + columnIndex + " to time: " + e, e);
+      }
+    }
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java:504]:
```suggestion
    if (value == null) {
      return null;
    }
    try {
      return parseTime(value, cal);
    } catch (IllegalArgumentException e) {
      throw new KyuubiSQLException("Cannot convert column " + columnIndex + " to time: " + e, e);
    }
```

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java [hunk 4]
```diff
@@ -412,6 +440,85 @@ public Timestamp getTimestamp(String columnName) throws SQLException {
...
+    if (value == null) {
+      return null;
+    } else {
+      try {
+        return parseTimestamp(value, cal);
+      } catch (IllegalArgumentException e) {
+        throw new KyuubiSQLException(
+            "Cannot convert column " + columnIndex + " to timestamp: " + e, e);
+      }
+    }
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java:455]:
```suggestion
    if (value == null) {
      return null;
    }
    try {
      return parseTimestamp(value, cal);
    } catch (IllegalArgumentException e) {
      throw new KyuubiSQLException(
          "Cannot convert column " + columnIndex + " to timestamp: " + e, e);
    }
```

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java [hunk 5]
```diff
@@ -182,6 +183,33 @@ public Date getDate(String columnName) throws SQLException {
...
+    Date value = getDate(columnIndex);
+    if (value == null) {
+      return null;
+    } else {
+      try {
+        return parseDate(value, cal);
+      } catch (IllegalArgumentException e) {
+        throw new KyuubiSQLException("Cannot convert column " + columnIndex + " to date: " + e, e);
+      }
+    }
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java:197]:
```suggestion
    if (value == null) {
      return null;
    }
    try {
      return parseDate(value, cal);
    } catch (IllegalArgumentException e) {
      throw new KyuubiSQLException("Cannot convert column " + columnIndex + " to date: " + e, e);
    }
```

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiArrowBasedResultSet.java [hunk 6]
```diff
@@ -406,6 +434,85 @@ public Timestamp getTimestamp(String columnName) throws SQLException {
...
+    Time value = getTime(columnIndex);
+    if (value == null) {
+      return null;
+    } else {
+      try {
+        return parseTime(value, cal);
+      } catch (IllegalArgumentException e) {
+        throw new KyuubiSQLException("Cannot convert column " + columnIndex + " to time: " + e, e);
+      }
+    }
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiArrowBasedResultSet.java:498]:
```suggestion
    }
    try {
      return parseTime(value, cal);
    } catch (IllegalArgumentException e) {
      throw new KyuubiSQLException("Cannot convert column " + columnIndex + " to time: " + e, e);
    }
```

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiArrowBasedResultSet.java [hunk 7]
```diff
@@ -406,6 +434,85 @@ public Timestamp getTimestamp(String columnName) throws SQLException {
...
+    if (value == null) {
+      return null;
+    } else {
+      try {
+        return parseTimestamp(value, cal);
+      } catch (IllegalArgumentException e) {
+        throw new KyuubiSQLException(
+            "Cannot convert column " + columnIndex + " to timestamp: " + e, e);
+      }
+    }
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiArrowBasedResultSet.java:449]:
```suggestion
    }
    try {
      return parseTimestamp(value, cal);
    } catch (IllegalArgumentException e) {
      throw new KyuubiSQLException(
          "Cannot convert column " + columnIndex + " to timestamp: " + e, e);
    }
```

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiArrowBasedResultSet.java [hunk 8]
```diff
@@ -198,6 +199,33 @@ public Date getDate(String columnName) throws SQLException {
...
+    Date value = getDate(columnIndex);
+    if (value == null) {
+      return null;
+    } else {
+      try {
+        return parseDate(value, cal);
+      } catch (IllegalArgumentException e) {
+        throw new KyuubiSQLException("Cannot convert column " + columnIndex + " to date: " + e, e);
+      }
+    }
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiArrowBasedResultSet.java:213]:
```suggestion
    }
    try {
      return parseDate(value, cal);
    } catch (IllegalArgumentException e) {
      throw new KyuubiSQLException("Cannot convert column " + columnIndex + " to date: " + e, e);
    }
```

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master/1.8

**Pull request closed without merging**